src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -415,32 +419,32 @@ public static unsafe void AddScalarU(float scalar, Span<float> dst)
         {
             fixed (float* pdst = dst)
             {
-                float* pDstEnd = pdst + dst.Length;
-                float* pDstCurrent = pdst;
-
                 Vector256<float> scalarVector256 = Avx.SetAllVector256(scalar);
+                int countAvx = Math.DivRem(dst.Length, AvxAlignment, out int remainderAvx);
+                float* pDstCurrent = pdst;
 
-                while (pDstCurrent + 8 <= pDstEnd)
+                for (int i = 0; i < countAvx; i++)
                 {
                     Vector256<float> dstVector = Avx.LoadVector256(pDstCurrent);
                     dstVector = Avx.Add(dstVector, scalarVector256);
                     Avx.Store(pDstCurrent, dstVector);
 
-                    pDstCurrent += 8;
+                    pDstCurrent += AvxAlignment;
                 }
 
                 Vector128<float> scalarVector128 = Sse.SetAllVector128(scalar);
+                int countSse = Math.DivRem(remainderAvx, SseIntrinsics.SseAlignment, out int remainderSse);
```

Comment:
You shouldn't need a `countSse`. At this point, we have somewhere between `0` and `7` elements remaining, so you can just check `if (remainder >= 4)`